core/src/com/unciv/ui/utils/ExtensionFunctions.kt

```diff
@@ -196,18 +184,57 @@ fun Float.toPercent() = 1 + this/100
 /** Translate a [String] and make a [TextButton] widget from it */
 fun String.toTextButton() = TextButton(this.tr(), BaseScreen.skin)
 
-/** Translate a [String] and make a [Button] widget from it, with control over font size, font colour, and an optional icon. */
-fun String.toButton(fontColor: Color = Color.WHITE, fontSize: Int = 24, icon: String? = null): Button {
+/**
+ * Builder-like configuration scope for [String.toButton].
+ *
+ * Properties: [button], [icon], [label], [iconCell], [labelCell]
+ *
+ * @param I [Actor] subclass of the icon given to [String.toButton].
+ * */
+interface IIconAndLabelCellsInButton<I: Actor?> {
```

Comment:
There is no need for this to be an interface if the only usage is to generate a single-instance class out of it.
Also, I doubt we're actually using all these cells. Please clean up all fields that are not in actual use - we can always add them later if needed.
The most maintainable code is that which isn't written :)

Reply:
Also, do we actually use any non-Actor functions of the icon anywhere? If not then we don't even need the templating. 
This is starting to look more like `IconLabelButton : Button(BaseScreen.skin){...}` which is much simpler.